--- uart_linux.c	2014-03-05 04:03:13.000000000 -0800
+++ uart_linux_new.c	2016-05-26 20:28:35.000000000 -0700
@@ -18,7 +18,7 @@
    int fd;
    struct termios options;
 
-   fd = open(ttyDev, O_RDWR | O_NOCTTY | O_NDELAY);
+   fd = open(ttyDev, O_RDWR | O_NOCTTY); // | O_NDELAY);
    if (fd <0)
    {
       return -1;
@@ -27,8 +27,8 @@
    fcntl(fd, F_SETFL, 0);
    tcgetattr(fd,&options); /* get current port settings */
 
-   cfsetispeed(&options, B115200);
-   cfsetospeed(&options, B115200);
+   cfsetispeed(&options, 115200);
+   cfsetospeed(&options, 115200);
 
    // 8-bit chars, 2 stop bits for ZigBee
    options.c_cflag = (options.c_cflag & ~CSIZE) | CS8 | CSTOPB /*| CRTSCTS */;
@@ -46,7 +46,7 @@
    options.c_cflag |= (CLOCAL | CREAD);// ignore modem controls,
                                   // enable reading
    options.c_cflag &= ~(PARENB | PARODD);      // shut off parity
-   options.c_cflag &= ~CRTSCTS;
+   //options.c_cflag &= ~CRTSCTS;
 
    if (tcsetattr (fd, TCSANOW, &options) != 0)
    {
